xbmc/guilib/GUISpinControlEx.cpp

```diff
@@ -136,9 +136,26 @@ void CGUISpinControlEx::SetSpinPosition(float spinPosX)
 
 void CGUISpinControlEx::RenderText(float posX, float posY, float width, float height)
 {
-  const float spaceWidth = 10;
-  // check our limits from the button control
-  float x = std::max(m_buttonControl.m_label.GetRenderRect().x2 + spaceWidth, posX);
+  const float freeSpaceWidth{m_buttonControl.GetWidth() - GetSpinWidth() * 2};
+
+  // Limit right label text width to max 50% of free space
+  // (will be slightly shifted due to offsetX padding)
+  const float rightTextMaxWidth{freeSpaceWidth * 0.5f};
+
+  float rightTextWidth{width};
+  if (rightTextWidth > rightTextMaxWidth)
+  {
+    rightTextWidth = rightTextMaxWidth - m_label.GetLabelInfo().offsetX;
+  }
+
   m_label.SetScrolling(HasFocus());
-  CGUISpinControl::RenderText(x, m_buttonControl.GetYPosition(), width + posX - x, m_buttonControl.GetHeight());
+  // Replace posX by using our button position
+  posX = m_buttonControl.GetXPosition() + freeSpaceWidth - rightTextWidth -
+         m_label.GetLabelInfo().offsetX;
+
+  CGUISpinControl::RenderText(posX, m_buttonControl.GetYPosition(), rightTextWidth,
+                              m_buttonControl.GetHeight());
+
+  // Limit the max width for the left label to avoid text overlapping
```

Comment:
Can you explain why the call to `SetMaxWidth` has to be done before calling into RenderText? Does RenderText update some of the control member variables such has offsetX? If they are not directly dependent can the `SetMaxWidth` be moved before the RenderText call?
I mean, to me it makes sense to:
- Do any sort of calculation for size/offsets
- Change stuff
- Render()

If for some reason we need to render, and then get information for resize it could be pointing out some issue in the logic/encapsulation

Reply:
yes i can use that points order,
just move `m_buttonControl.Render();` before `CGUISpinControl::RenderText`
but i need to change a little thing in CGUISpinControl to render spin buttons after textrender and not before
otherwise `m_buttonControl.Render();` clear the buttons

Reply:
pushed fixup